### ISSUE TITLE
JS - UnusedPrivateMember: unused parameter in constructor of external classes

### ISSUE THREAD

**Issue by hfhbd** [detekt/detekt#4913]
## Expected Behavior
No issue

## Observed Behavior
`untitled9/src/main/kotlin/Foo.kt:3:20: Private property `foo` is unused. [UnusedPrivateMember]`

## Steps to Reproduce
```kotlin
@JsModule("foo")
@JsNonModule
external class Foo(foo: String)
```

`./gradlew detekt`

## Context
I want to use an external class from a npm module.

## Tidbit
I added this test to `UnusedPrivateMemberSpec`, which does not trigger an issue. Maybe the issue is JS specific, and the test uses JVM, doesn't it?
```kotlin
@Test
fun `should not report constructor parameters in external classes`() {
    val code = """
        external class Foo(foo: String)
    """
    assertThat(subject.lint(code).isEmpty())
}
```

## Your Environment

* Version of detekt used: 1.20
* Version of Gradle used (if applicable): 7.4.2
* Gradle scan link (add `--scan` option when running the gradle task):
* Operating System and version: macOS 12
* Link to your project (if it's a public repository): [untitled9.zip](https://github.com/detekt/detekt/files/8837739/untitled9.zip)

**Comment by cortinico** [comment]
I'm not sure what's going on here. I can clearly see that the rule has logic to handle `external` function:

https://github.com/detekt/detekt/blob/9146e4d1f65034e45edd017b8e2dc0d84062fc8f/detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/UnusedPrivateMember.kt#L271-L273

The JS annotations are not relevant for the sake of this inspection.

> I added this test to `UnusedPrivateMemberSpec`, which does not trigger an issue. Maybe the issue is JS specific, and the test uses JVM, doesn't it?

Would you be fine adding this test case to the detekt codebase?

**Comment by hfhbd** [comment]
Yes, I saw the logic too. 

I simple added the test but this didn't work, the test succeeded. Don't know, what is wrong.

**Comment by cortinico** [comment]
It might be that you're not running with type resolution

**Comment by hfhbd** [comment]
Yeah, type resolution is only supported on the JVM, isn't it?

**Comment by cortinico** [comment]
Yeah that's correct 🤔 Still this section on the rule should be fine with just syntax-level rule. It's still worth investigating what's going on.

**Comment by hfhbd** [comment]
still relevant

**Comment by YoshiRulz** [comment]
Parameters of external class constructors still flagged on 1.22.0.
